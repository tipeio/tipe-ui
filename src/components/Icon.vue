<template>
  <div
    :class="size"
    :data-tipe-ui="$options.name"
  />
</template>

<script>
import vueTypes from 'vue-types'

const sizes = ['small', 'medium', 'large']

export default {
  name: 'TipeIcon',
  props: {
    size: vueTypes.oneOf(sizes).def('medium')
  }
}
</script>

<style scoped>
[data-tipe-ui='TipeIcon'] {
  background-color: red;
}

[data-tipe-ui='TipeIcon'].small {
  height: 1rem;
  width: 1rem;
}

[data-tipe-ui='TipeIcon'].medium {
  height: 2rem;
  width: 2rem;
}

[data-tipe-ui='TipeIcon'].large {
  height: 3rem;
  width: 3rem;
}
</style>
