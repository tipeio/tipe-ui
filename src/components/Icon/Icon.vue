<template>
  <object
    :data="src"
    type="image/svg+xml">
    <img :src="src" >
  </object>
</template>

<script>
import vueTypes from 'vue-types'

export default {
  name: 'TipeIcon',
  props: {
    icon: vueTypes.string
  },
  computed: {
    src() {
      return require(`../../icons/${this.icon}.svg`)
    }
  }
}
</script>
