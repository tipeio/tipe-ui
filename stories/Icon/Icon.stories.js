import { storiesOf } from '@storybook/vue'

import TipeIcon from '../../src/components/Icon'

const style = () => ({
  height: '1rem',
  width: '1rem'
})

const icons = [
  'check',
  'document',
  'folder',
  'key',
  'number',
  'share',
  'switch',
  'user',
  'color',
  'edit',
  'gear',
  'link',
  'pin',
  'sign-out',
  'text',
  'view',
  'copy',
  'flow',
  'grid',
  'list',
  'search',
  'star-active',
  'trash',
  'wrench',
  'credit-card',
  'folder-open',
  'group',
  'markdown',
  'media',
  'send',
  'star',
  'user-circle',
  'options',
  'plus',
  'cross',
  'arrow-down',
  'arrow-up',
  'arrow-right',
  'arrow-left'
]

storiesOf('Icon', module).add('icons', () => ({
  components: { TipeIcon },
  computed: { style },
  template:
    '<div>' +
    icons.reduce((template, icon) => {
      return (
        template +
        `<tipe-icon :style="style" icon="${icon}" color="rgb(255,0,0)" />`
      )
    }, '') +
    '</div>'
}))
