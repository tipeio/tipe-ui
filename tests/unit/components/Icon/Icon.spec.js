import { shallowMount } from '@vue/test-utils'
import Icon from '@/components/Icon'

describe('Icon', () => {
  it('matches previous snapshot', () => {
    const propsData = { icon: 'folder' }
    const wrapper = shallowMount(Icon, { propsData })

    expect(wrapper).toMatchSnapshot()
  })

  it('has correct data-tipe-iu attibute', () => {
    const propsData = { icon: 'folder' }
    const wrapper = shallowMount(Icon, { propsData })

    expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeIcon')
  })
})
