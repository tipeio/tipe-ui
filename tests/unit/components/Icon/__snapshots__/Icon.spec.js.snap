// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon <template> matches previous snapshot 1`] = `
<svg data-tipe-ui="TipeIcon" viewBox="0 0 20 20" fill="#000" style="height: 1rem; width: 1rem;">
  <use xlink:href="#test"></use>
</svg>
`;
